packages/transaction-messages/src/compile/message.ts

```diff
@@ -58,26 +66,29 @@ type VersionedCompiledTransactionMessage = BaseCompiledTransactionMessage &
  * @see {@link decompileTransactionMessage}
  */
 export function compileTransactionMessage(
-    transactionMessage: CompilableTransactionMessage & Readonly<{ version: 'legacy' }>,
+    transactionMessage: BaseTransactionMessage & Readonly<{ version: 'legacy' }> & TransactionMessageWithFeePayer,
 ): LegacyCompiledTransactionMessage;
 export function compileTransactionMessage(
-    transactionMessage: CompilableTransactionMessage,
+    transactionMessage: BaseTransactionMessage & TransactionMessageWithFeePayer,
 ): VersionedCompiledTransactionMessage;
 export function compileTransactionMessage(
-    transactionMessage: CompilableTransactionMessage,
+    transactionMessage: BaseTransactionMessage & TransactionMessageWithFeePayer,
 ): CompiledTransactionMessage {
     const addressMap = getAddressMapFromInstructions(
         transactionMessage.feePayer.address,
         transactionMessage.instructions,
     );
     const orderedAccounts = getOrderedAccountsFromAddressMap(addressMap);
+    const lifetimeConstraint =
+        (transactionMessage as Partial<TransactionMessageWithLifetime>).lifetimeConstraint ??
+        EMPTY_BLOCKHASH_LIFETIME_CONSTRAINT;
     return {
         ...(transactionMessage.version !== 'legacy'
             ? { addressTableLookups: getCompiledAddressTableLookups(orderedAccounts) }
             : null),
         header: getCompiledMessageHeader(orderedAccounts),
         instructions: getCompiledInstructions(transactionMessage.instructions, orderedAccounts),
-        lifetimeToken: getCompiledLifetimeToken(transactionMessage.lifetimeConstraint),
+        lifetimeToken: getCompiledLifetimeToken(lifetimeConstraint),
```

Comment:
Honestly, I thought we'd just make this optional on `CompiledTransactionMessage` and then make the _compiler_ (ie. `getCompiledTransactionMessageEncoder()`) encapsulate the behaviour of filling it in when absent.

Actually I think I like that better, because then we're not passing a value with a `"11111111111111111111111111111111"` in it around your app.